File: src/BaseHmiTypes/Screens/Shapes/HmiRectangle.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public class HmiRectangle : HmiSurfaceShapeBase
{
    public HmiRectangle()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiRectangle;
    }

    public HmiProperty<(double x, double y)>? TopLeftRadius { get; set; }

    public HmiProperty<(double x, double y)>? TopRightRadius { get; set; }

    public HmiProperty<(double x, double y)>? BottomLeftRadius { get; set; }

    public HmiProperty<(double x, double y)>? BottomRightRadius { get; set; }

    public HmiProperty<int>? CornerRadius { get; set; }

    public HmiProperty<int>? CornerStyle { get; set; }

    public HmiProperty<int>? EdgeStyle { get; set; }
}

File: typescript/base-hmi-types/src/screens/shapes/HmiRectangle.ts
import { HmiProperty } from "../base.js";
import { HmiSurfaceShapeBase } from "./HmiSurfaceShapeBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiRectangle extends HmiSurfaceShapeBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiRectangle;
  }

  topLeftRadius?: HmiProperty<{ x: number; y: number }>;
  topRightRadius?: HmiProperty<{ x: number; y: number }>;
  bottomLeftRadius?: HmiProperty<{ x: number; y: number }>;
  bottomRightRadius?: HmiProperty<{ x: number; y: number }>;
  cornerRadius?: HmiProperty<number>;
  cornerStyle?: HmiProperty<number>;
}
